src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -262,7 +262,7 @@ protected override ImmutableArray<PendingBranch> RemoveReturns()
                 var foundAwait = result.Any(pending => pending.Branch?.Kind == BoundKind.AwaitExpression);
                 if (!foundAwait)
                 {
-                    Diagnostics.Add(ErrorCode.WRN_AsyncLacksAwaits, currentMethodOrLambda.Locations[0]);
+                    Diagnostics.Add(ErrorCode.WRN_AsyncLacksAwaits, currentMethodOrLambda.NarrowLocation());
```

Comment:
Given that we've been explicit that the compiler will not guarantee diagnostic location or ID, then yes i'm ok with that.